Makefile

```diff
@@ -0,0 +1,138 @@
+# Common definitions ############################
+
+# Declare "all" as the default target that is built when "make" is run without
+# extra arguments.
+all:
+.PHONY: all
+
+
+# Toolchain-agnostic targets ####################
+
+TARGETS := \
+	common/cpp/build \
+	example_cpp_smart_card_client_app/build \
+	example_js_smart_card_client_app/build \
+	example_js_standalone_smart_card_client_library \
+	third_party/pcsc-lite/naclport/common/build \
+	third_party/pcsc-lite/naclport/cpp_client/build \
+	third_party/pcsc-lite/naclport/cpp_demo/build \
+
+example_cpp_smart_card_client_app/build: common/cpp/build
+example_cpp_smart_card_client_app/build: third_party/pcsc-lite/naclport/common/build
+example_cpp_smart_card_client_app/build: third_party/pcsc-lite/naclport/cpp_client/build
+example_cpp_smart_card_client_app/build: third_party/pcsc-lite/naclport/cpp_demo/build
+
+TEST_TARGETS := \
+	common/cpp/build/tests \
+	common/js/build/unittests \
+	third_party/libusb/naclport/build/js_unittests \
+	third_party/pcsc-lite/naclport/server_clients_management/build/js_unittests \
+
+common/cpp/build/tests: common/cpp/build
+
+# Toolchain related definitions #################
+
+TOOLCHAIN ?= pnacl
+
+ifeq ($(TOOLCHAIN),emscripten)
+
+# Emscripten-specific definitions ###############
+
+TARGETS += \
+	third_party/googletest/webport/build \
+
+common/cpp/build/tests: third_party/googletest/webport/build
+
+else ifeq ($(TOOLCHAIN),pnacl)
+
+# Native Client specific definitions ############
+
+TARGETS += \
+	smart_card_connector_app/build \
+	third_party/ccid/naclport/build \
+	third_party/libusb/naclport/build \
+	third_party/pcsc-lite/naclport/server/build \
+	third_party/pcsc-lite/naclport/server_clients_management/build \
+
+smart_card_connector_app/build: common/cpp/build
+smart_card_connector_app/build: third_party/ccid/naclport/build
+smart_card_connector_app/build: third_party/libusb/naclport/build
+smart_card_connector_app/build: third_party/pcsc-lite/naclport/common/build
+smart_card_connector_app/build: third_party/pcsc-lite/naclport/server/build
+smart_card_connector_app/build: third_party/pcsc-lite/naclport/server_clients_management/build
+third_party/ccid/naclport/build: third_party/libusb/naclport/build
+third_party/pcsc-lite/naclport/server/build: third_party/libusb/naclport/build
+third_party/pcsc-lite/naclport/server_clients_management/build: third_party/pcsc-lite/naclport/server/build
+
+TEST_TARGETS += \
+	common/integration_testing/build \
+	example_cpp_smart_card_client_app/build/integration_tests \
+	third_party/libusb/naclport/build/tests \
+
+example_cpp_smart_card_client_app/build/integration_tests: common/cpp/build
+example_cpp_smart_card_client_app/build/integration_tests: common/integration_testing/build
+example_cpp_smart_card_client_app/build/integration_tests: example_cpp_smart_card_client_app/build
+third_party/libusb/naclport/build/tests: common/cpp/build
+third_party/libusb/naclport/build/tests: third_party/ccid/naclport/build
```

Comment:
self-nit: The prerequisite actually needs to be "third_party/libusb/naclport/build".